### PR TITLE
doc: Test for broken URLs in `manpage-urls.json`

### PR DESCRIPTION
**Pull request by nicoonoclaste**
## Description of changes

- [x] Add `maintainers/scripts/doc/check-manpage-urls.py`
- [x] Fix broken manpage URLs
- [x] Document the script & its usage in `maintainers/scripts/README.md`

## Things done

- [x] Successfully run `check-manpage-urls.py` (on `x86_64-linux`)
- [x] Check with linter (`ruff`) and gradual type checker (`mypy`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: doc/manpage-urls.json
Comment (jtojnar) [inline doc/manpage-urls.json:2]: https://github.com/NixOS/nixpkgs/blob/fcd2f54d7d01c44bef3512f29c2c6969b0c9f80e/nixos/modules/services/networking/gnunet.nix#L124

**Comment by mkg20001** [comment]
can you rebase to fix ci?

File: maintainers/scripts/doc/README.md
Comment (infinisil) [inline maintainers/scripts/doc/README.md:1]: Since recently there's already `maintainers/scripts/README.md`, so that should be used instead now.

**Review by infinisil — CHANGES_REQUESTED** [review]

**Review by infinisil — COMMENTED** [review]
How about automating this as part of the docs build? It already runs [for every PR that updates the docs](https://github.com/NixOS/nixpkgs/blob/master/.github/workflows/manual-nixpkgs.yml), and this check could be done similarly to the [maintainer GitHub id check](https://github.com/NixOS/nixpkgs/blob/master/lib/tests/maintainers.nix). This could be done in `doc/default.nix`.

File: doc/default.nix
Comment (infinisil) [reply doc/default.nix:120]: Thinking about this some more, this sure is impure, so I don't think the real docs derivation should be "tainted" by this. So I'd suggest to put this in `passthru.tests.checkManpageUrls` instead. The github workflow to build the manual can then be adjusted to additionally build this attribute:

```diff
diff --git a/.github/workflows/manual-nixpkgs.yml b/.github/workflows/manual-nixpkgs.yml
index d10b5b9bb701..8d4e68d525cd 100644
--- a/.github/workflows/manual-nixpkgs.yml
+++ b/.github/workflows/manual-nixpkgs.yml
@@ -29,4 +29,4 @@ jobs:
           name: nixpkgs-ci
           signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
       - name: Building Nixpkgs manual
-        run: NIX_PATH=nixpkgs=$(pwd) nix-build --option restrict-eval true pkgs/top-level/release.nix -A manual
+        run: NIX_PATH=nixpkgs=$(pwd) nix-build --option restrict-eval true pkgs/top-level/release.nix -A manual -A manual.checkManpageUrls
```

File: maintainers/scripts/doc/check-manpage-urls.py
Comment (infinisil) [inline maintainers/scripts/doc/check-manpage-urls.py:1]: Oh and if we run it in CI, I guess this doesn't need to be a maintainer script. Probably `pkgs/test/check-manpage-urls/run.py` would be a good place, you can put the docs in a `README.md` next to it. Though we probably don't want to add it to `pkgs/test/default.nix`, because then it would get built by Hydra (since it's impure).

I'm not very sure though, Nixpkgs needs a better structure for this, I'll request some reviews by others.

**Review by infinisil — COMMENTED** [review]
Looking pretty good! I'll request some reviews by people that could give some good feedback too

File: maintainers/scripts/doc/check-manpage-urls.py
Comment (infinisil) [reply maintainers/scripts/doc/check-manpage-urls.py:1]: Sounds fine to me!

**Review by pennae — APPROVED** [review]

**Review by infinisil — APPROVED** [review]

File: doc/tests/manpage-urls.py
Comment (Mic92) [inline doc/tests/manpage-urls.py:71]: I think I'll add some more types here, but this can also happen after the merge.

File: doc/tests/manpage-urls.py
Comment (Mic92) [inline doc/tests/manpage-urls.py:54]: Is there an upper bound on how many concurrent connections this will fire at once?

**Review by infinisil — APPROVED** [review]

File: doc/tests/manpage-urls.py
Comment (Mic92) [reply doc/tests/manpage-urls.py:55]: sounds reasonable.

**Review by Mic92 — APPROVED** [review]

**Pull request merged**
